src/main/java/de/hpi/bpt/argos/api/entityMapping/EntityMappingEndpointImpl.java

```diff
@@ -0,0 +1,238 @@
+        if (mapping == null) {
+            halt(HttpStatusCodes.BAD_REQUEST, "given mapping was not found");
+        }
+
+        // delete corresponding mapping conditions
+        List<MappingCondition> conditions = PersistenceAdapterImpl.getInstance().getMappingConditions(mappingId);
+        if (!PersistenceAdapterImpl.getInstance().deleteArtifacts(conditions.toArray(new MappingCondition[conditions.size()]))) {
+            halt(HttpStatusCodes.ERROR, "could not delete mapping conditions");
+        }
+
+        if (!PersistenceAdapterImpl.getInstance().deleteArtifacts(mapping)) {
+            halt(HttpStatusCodes.ERROR, "could not delete mapping");
+        }
+
+        endpointUtil.logSendingResponse(logger, request, response.status(), response.body());
+        return response.body();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String editEntityMapping(Request request, Response response) {
+        endpointUtil.logReceivedRequest(logger, request);
+
+        JsonObject jsonBody = new JsonObject();
+        try {
+            jsonBody = jsonParser.parse(request.body()).getAsJsonObject();
+        } catch (JsonSyntaxException e) {
+            LoggerUtilImpl.getInstance().error(logger, JSON_PARSE_ERROR_MESSAGE, e);
+            halt(HttpStatusCodes.BAD_REQUEST, "not a valid json");
+        }
+
+        JsonObject jsonMapping = jsonBody.get(JSON_ENTITY_MAPPING_ATTRIBUTE).getAsJsonObject();
+        if (jsonMapping == null) {
+            halt(HttpStatusCodes.BAD_REQUEST, "no event mapping given in body");
+        }
+
+        EventEntityMapping oldMapping = PersistenceAdapterImpl.getInstance().getEventEntityMapping(getMappingId(request));
+
+        // update conditions
+        List<MappingCondition> oldConditions = PersistenceAdapterImpl.getInstance().getMappingConditions(oldMapping.getId());
+        if (!PersistenceAdapterImpl.getInstance().deleteArtifacts(oldConditions.toArray(new MappingCondition[oldConditions.size()]))) {
+            halt(HttpStatusCodes.ERROR, "could not delete conditions");
+        }
+
+        List<MappingCondition> newConditions = getConditionsFromJson(jsonMapping.get(MAPPING_CONDITIONS_ATTRIBUTE).getAsJsonArray(), oldMapping.getId());
+        if (!PersistenceAdapterImpl.getInstance().saveArtifacts(newConditions.toArray(new MappingCondition[newConditions.size()]))) {
+            halt(HttpStatusCodes.ERROR, "could not create new conditions");
+        }
+
+        // update mapping
+        EventEntityMapping newMapping = getEventMappingFromJson(jsonMapping);
+        oldMapping.setEventTypeId(newMapping.getEventTypeId());
+        oldMapping.setEntityTypeId(newMapping.getEntityTypeId());
+        oldMapping.setTargetStatus(newMapping.getTargetStatus());
+
+        if (!PersistenceAdapterImpl.getInstance().updateArtifact(oldMapping, EventTypeEndpoint.getEventTypeEntityMappingsUri(oldMapping.getEventTypeId()))) {

```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Split this 157 characters long line (which is greater than 150 authorized). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://bpt-lab.org/sonarqube/coding_rules#rule_key=squid%3AS00103)